designs/data_serialization_design.md

```diff
@@ -0,0 +1,189 @@
+# FAIR Data Serialization Design
```

Comment:
This is a good design template. Do you mind in a separate PR adding a design_template.md in the same directory for the future designers to reference so we have a standard template?

Reply:
Created an issue.
https://github.com/satmihir/fair/issues/45